app/javascript/mastodon/api.ts

```diff
@@ -83,6 +83,7 @@ export default function api(withAuthorization = true) {
   return instance;
 }
 
+type ApiUrl = `v${1 | 2}/${string}`;
```

Comment:
I wonder how much of a good idea this is. We also use `/api/web` (through none of the Typescript API paths yet) and in the past we had `/api/v2_alpha` for a pre-release feature. We'll likely have a `v3` at some point as well. But I suppose we can change `ApiUrl` when needed, if needed.

Reply:
This will only throw TypeScript errors, not prevent the code from running, so I think this is OK and can be changed when more paths are added to TypeScript.

Reply:
I wasn't 100% sure about this either @ClearlyClaire. But if it turns out to be annoying or limiting, it'll also be easy enough to revert and replace with `url.startsWith('/') ? url.slice(1) : url` in the function while allowing leading slashes in the function call.